### PR TITLE
Update Prerequisite.md

### PR DESCRIPTION
**Pull request by IshaanAdarsh**
The PR is for reviewing the changes and suggest any for improving this part of the project

### CODE REVIEW

File: Quick-Start/Prerequisite.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ The extension architecture follows a modular approach, where each extension is p
     - Extensions can register hooks, define custom functions, operators, and data types, and override or extend PostgreSQL's behaviour.
     - The Extension API enables seamless integration of the extension's functionality with the core database system.
 
-5. **Installation and Management:**
+5. **Installation and Management:** 
     - Extensions are installed using SQL commands, such as `CREATE EXTENSION` or `ALTER EXTENSION`, which reference the extension's control file.
```
Comment (theory) [inline Quick-Start/Prerequisite.md]:
```suggestion
    - Extensions are installed using the SQL command `CREATE EXTENSION`, and updated with the `ALTER EXTENSION` command, both of which reference the extension's control file. 
```

File: Quick-Start/Prerequisite.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ PostgreSQL supports various target languages for extension development. Here is
 
 ## C:
```
Comment (theory) [inline Quick-Start/Prerequisite.md]: I suggest starting with SQL, then PL/pgSQL, and then C and the other languages. Start with the simplest first.

File: Quick-Start/Prerequisite.md [hunk 3]
```diff
@@ -60,7 +60,7 @@ PostgreSQL supports various target languages for extension development. Here is
...
 The C programming language is widely used for developing PostgreSQL extensions. It offers direct access to PostgreSQL internals and provides high performance.
- - Development Environment: To develop C-based extensions for PostgreSQL, you need a C compiler (such as GCC) and the necessary development libraries for PostgreSQL. 
+ - Development Environment: To develop C-based extensions for PostgreSQL, you need a C compiler (such as GCC) and the necessary development libraries for PostgreSQL.  
    - On Linux systems, you may need to install the `postgresql-server-dev` package. 
    - On macOS, you can use Xcode Command Line Tools or install PostgreSQL via Homebrew. 
    - On Windows, you can use Microsoft Visual Studio or MinGW.
```
Comment (theory) [inline Quick-Start/Prerequisite.md]: Other comments:

* You do not need DBD::Pg to use PL/Perl, but keep the notes about installing stuff from CPAN.
* I suggest you also mention PL/PerlU.
* I don't believe Ruby extensions need `pg` anymore than Perl extensions need DBD::Pg, but you should verify that.
* Same for Java: I don't believe you need the JDBC driver.
*

**Review by theory — APPROVED** [review]
Great start!

File: Quick-Start/Prerequisite.md (hunk 3 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md]: I'll update these immediately, my source material was faulty

File: Quick-Start/Prerequisite.md (hunk 1 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md]: done

File: Quick-Start/Prerequisite.md (hunk 2 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md]: I have just listed these out to inform the user about the various target languages which are supported for extension development.
I'll only be working on a selected few languages in this tutorial.

Comment (theory) [reply Quick-Start/Prerequisite.md]: Sure, I'm just suggesting to list them in order of complexity to use.

File: Quick-Start/Prerequisite.md [hunk 4]
```diff
@@ -1,101 +1,101 @@
...
-## JavaScript:
-PostgreSQL supports JavaScript through the `plv8` extension. It allows you to write extension functions using JavaScript, leveraging the power of JavaScript frameworks and libraries.
- - Development Environment: JavaScript-based extensions using the `plv8` extension require the V8 JavaScript engine and the necessary development libraries. The `plv8` extension provides installation instructions for the required dependencies.
-<!-- Need to add pl/js, but it is in the initial stages (ask David and Jimmy)-->
+# What is a Postgres extension and its purpose: 
+ 
+A PostgreSQL extension is a modular unit that extends the functionality of the PostgreSQL database system. It allows you to add custom features, data types, functions, operators, and other database objects to tailor PostgreSQL to specific needs and requirements.  
+
+PostgreSQL extensions provide a way to package and distribute additional functionality for PostgreSQL in a standardized and manageable manner. Here are some key points about the purpose of PostgreSQL extensions: 
+
```
Comment (theory) [inline Quick-Start/Prerequisite.md:6]: Not sure why you have trailing spaces on every line, but your editor should have some command to clean them up.

File: Quick-Start/Prerequisite.md [hunk 5]
```diff
@@ -1,101 +1,101 @@
...
+ 
+- **Community Contributions:** PostgreSQL has a vibrant and active community that develops and maintains a wide range of extensions. These extensions cover various domains such as GIS, full-text search, JSON processing, time-series data, and more. By leveraging existing extensions, developers can benefit from community expertise and avoid reinventing the wheel. 
+ 
+- **Performance Optimization:** Extensions can also be used to optimize the performance of your PostgreSQL database. They can provide advanced indexing techniques, caching mechanisms, or query optimization algorithms that improve query execution speed and overall database performance.  
+ 
+Overall, PostgreSQL extensions enhance the flexibility, customization, and extensibility of the PostgreSQL database system. They allow developers and users to tailor the database to their specific needs, leverage community-contributed functionality, and simplify the installation and management of additional features.  
+# Basic Architecture:
```
Comment (theory) [inline Quick-Start/Prerequisite.md:22]: Generally you need a blank line on either side of a Markdown header to ensure they're formatted properly.

File: Quick-Start/Prerequisite.md [hunk 6]
```diff
@@ -1,101 +1,101 @@
...
+ 
+- **Community Contributions:** PostgreSQL has a vibrant and active community that develops and maintains a wide range of extensions. These extensions cover various domains such as GIS, full-text search, JSON processing, time-series data, and more. By leveraging existing extensions, developers can benefit from community expertise and avoid reinventing the wheel. 
+ 
+- **Performance Optimization:** Extensions can also be used to optimize the performance of your PostgreSQL database. They can provide advanced indexing techniques, caching mechanisms, or query optimization algorithms that improve query execution speed and overall database performance.  
+ 
+Overall, PostgreSQL extensions enhance the flexibility, customization, and extensibility of the PostgreSQL database system. They allow developers and users to tailor the database to their specific needs, leverage community-contributed functionality, and simplify the installation and management of additional features.  
+# Basic Architecture:  
+The extension architecture follows a modular approach, where each extension is packaged as a separate unit. Let's explore the key components and how they work together:
```
Comment (theory) [inline Quick-Start/Prerequisite.md:23]:
```suggestion
The extension architecture follows a modular approach, where each extension is packaged as an independent unit. Let's explore the key components and how they work together: 
```

File: Quick-Start/Prerequisite.md [hunk 7]
```diff
@@ -1,101 +1,101 @@
...
+ 
+- **Performance Optimization:** Extensions can also be used to optimize the performance of your PostgreSQL database. They can provide advanced indexing techniques, caching mechanisms, or query optimization algorithms that improve query execution speed and overall database performance.  
+ 
+Overall, PostgreSQL extensions enhance the flexibility, customization, and extensibility of the PostgreSQL database system. They allow developers and users to tailor the database to their specific needs, leverage community-contributed functionality, and simplify the installation and management of additional features.  
+# Basic Architecture:  
+The extension architecture follows a modular approach, where each extension is packaged as a separate unit. Let's explore the key components and how they work together: 
+ 
+1. **Extension Control File:** 
+    - The extension control file, typically named `extension_name.control`, is a plain text file that provides metadata and instructions about the extension.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:26]: Suggest use a dollar sign to indicate variables, e.g., `$extension_name.control`.

File: Quick-Start/Prerequisite.md [hunk 8]
```diff
@@ -1,101 +1,101 @@
...
+ 
+Overall, PostgreSQL extensions enhance the flexibility, customization, and extensibility of the PostgreSQL database system. They allow developers and users to tailor the database to their specific needs, leverage community-contributed functionality, and simplify the installation and management of additional features.  
+# Basic Architecture:  
+The extension architecture follows a modular approach, where each extension is packaged as a separate unit. Let's explore the key components and how they work together: 
+ 
+1. **Extension Control File:** 
+    - The extension control file, typically named `extension_name.control`, is a plain text file that provides metadata and instructions about the extension. 
+    - It contains information such as the extension name, version, description, dependencies, installation script, and configuration options. 
+    - This file is used by PostgreSQL to manage the extension's installation, upgrade, and removal.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:28]: Maybe link to [the documentation](https://www.postgresql.org/docs/15/extend-extensions.html) for the control file.

File: Quick-Start/Prerequisite.md [hunk 9]
```diff
@@ -1,101 +1,101 @@
...
+    - It contains information such as the extension name, version, description, dependencies, installation script, and configuration options. 
+    - This file is used by PostgreSQL to manage the extension's installation, upgrade, and removal. 
+
+2. **Extension SQL File:** 
+    - The extension SQL file, typically named `extension_name.sql`, contains the SQL statements required to create the extension's objects in the database. 
+    - It can include table definitions, function definitions, views, triggers, and other database objects specific to the extension. 
+    - The SQL file is executed during the installation process to set up the extension's functionality in the database. 
+ 
+3. **Extension Library:** 
+    - Some extensions may require additional libraries or shared objects to provide their functionality.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:36]: So to be clear, you're talking about two different things here:

1. Some extensions are compiled into a shared library, generally when written in C. I think that's what you mean by the "Extension Library"
2. Some extensions when they compile their library include external libraries. These may be package with the extension or they might need to be installed before you can build the extension.

I think you're mainly talking about `#1` here; if so, omit the "additional libraries or shared objects" bit.

File: Quick-Start/Prerequisite.md [hunk 10]
```diff
@@ -1,101 +1,101 @@
...
+    - This file is used by PostgreSQL to manage the extension's installation, upgrade, and removal. 
+
+2. **Extension SQL File:** 
+    - The extension SQL file, typically named `extension_name.sql`, contains the SQL statements required to create the extension's objects in the database. 
+    - It can include table definitions, function definitions, views, triggers, and other database objects specific to the extension. 
+    - The SQL file is executed during the installation process to set up the extension's functionality in the database. 
+ 
+3. **Extension Library:** 
+    - Some extensions may require additional libraries or shared objects to provide their functionality. 
+    - The extension library is a compiled binary or a collection of source files that implement the extension's core functionality.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:37]: Well no, the library is compiled from a collection of one or more source files, which are generally written in C.

File: Quick-Start/Prerequisite.md [hunk 11]
```diff
@@ -1,101 +1,101 @@
...
+    - The extension SQL file, typically named `extension_name.sql`, contains the SQL statements required to create the extension's objects in the database. 
+    - It can include table definitions, function definitions, views, triggers, and other database objects specific to the extension. 
+    - The SQL file is executed during the installation process to set up the extension's functionality in the database. 
+ 
+3. **Extension Library:** 
+    - Some extensions may require additional libraries or shared objects to provide their functionality. 
+    - The extension library is a compiled binary or a collection of source files that implement the extension's core functionality. 
+    - The library is typically written in C or another programming language and interacts with PostgreSQL through the Extension API. 
+
+4. **Extension API:**
```
Comment (theory) [inline Quick-Start/Prerequisite.md:40]: I think you need a link here, but the API is not so much an extension API as the API of Postgres itself.

File: Quick-Start/Prerequisite.md [hunk 12]
```diff
@@ -1,101 +1,101 @@
...
+
+5. **Installation and Management:**  
+    - Extensions are installed using the SQL command `CREATE EXTENSION`, and updated with the `ALTER EXTENSION` command, both of which reference the extension's control file. 
+    - During installation, PostgreSQL reads the control file, validates dependencies, executes the installation script, and sets up the extension in the database. 
+    - Extensions can be managed using SQL commands to enable, disable, upgrade, or uninstall them. 
+
+6.  **Extension Upgrades:** 
+    - The extension control file specifies the version of the extension, allowing PostgreSQL to manage upgrades seamlessly. 
+    - When a new version of an extension is installed, PostgreSQL compares the version specified in the control file with the installed version. 
+    - If an upgrade script is provided, it is executed to migrate data or perform other necessary changes during the upgrade process.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:53]: Only by `ALTER EXTENSION UPDATE`.

File: Quick-Start/Prerequisite.md [hunk 13]
```diff
@@ -1,101 +1,101 @@
...
+The extension architecture provides a structured and standardized way to extend PostgreSQL's functionality. It allows developers to create reusable components, customize the database to specific needs, and integrate third-party functionality seamlessly. The modular design and integration with the Extension API ensure compatibility, ease of installation, and manageability of extensions within the PostgreSQL ecosystem. 
+
+# Supported Languages: 
+PostgreSQL supports various target languages for extension development. Here is a list of the supported target languages in Postgres Extension Creation: 
+
+## C: 
+The C programming language is widely used for developing PostgreSQL extensions. It offers direct access to PostgreSQL internals and provides high performance. 
+ - Development Environment: To develop C-based extensions for PostgreSQL, you need a C compiler (such as GCC) and the necessary development libraries for PostgreSQL.   
+   - On Linux systems, you may need to install the `postgresql-server-dev` package.  
+   - On macOS, you can use Xcode Command Line Tools or install PostgreSQL via Homebrew.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:64]: You need both the Xcode tools and the Postgres server.

File: Quick-Start/Prerequisite.md [hunk 14]
```diff
@@ -1,101 +1,101 @@
...
+ - Build System: C-based extensions typically use Makefiles or build systems like CMake or Autotools to compile and link the extension code. 
+
+## C++:  
+PostgreSQL also supports C++ for extension development. C++ allows object-oriented programming and provides additional features over the C language. 
+ - Development Environment: Similar to C, you need a C++ compiler and the development libraries for PostgreSQL. 
+ - Build System: `C++` extensions can use Makefiles or build systems like CMake or Autotools for compilation and linking. 
+
+## PL/pgSQL: 
+`PL/pgSQL` is the default procedural language of PostgreSQL. It is similar to `SQL` and allows you to write stored procedures, functions, and triggers using SQL-like syntax. 
+ - Development Environment: No additional development environment is required. You can write `PL/pgSQL` functions directly in the PostgreSQL database using SQL client tools.
```
Comment (theory) [inline Quick-Start/Prerequisite.md:75]: Yes, but ultimately you need to save your SQL to a file to be installed by the extension.

**Review by theory — CHANGES_REQUESTED** [review]

File: Quick-Start/Prerequisite.md (hunk 4 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md:6]: It's some issue, I checked it out. I has happened on each and every pull request form #3 to #8

File: Quick-Start/Prerequisite.md (hunk 5 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md:22]: I'll format every files after all the changes are done.

File: Quick-Start/Prerequisite.md (hunk 8 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md:28]: This is a doubt I had, so basically, Postgres is updated to newer versions every time. So I don't want my links to point to a previous version of postgres Doc, as there is a chance even in these 2 remaining months for the docs to change.

So I will edit all the links at the end of the project cycle.
But it is a good idea to comment something so I'll remember where the links are to be used

File: Quick-Start/Prerequisite.md (hunk 9 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md:36]: I will omit that part.

File: Quick-Start/Prerequisite.md (hunk 11 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md:40]: I will mention that the API refers to Postgres API. I will add a line explaining how it is used for the extension creation process

File: Quick-Start/Prerequisite.md (hunk 2 above)
Comment (IshaanAdarsh) [reply Quick-Start/Prerequisite.md]: Oh !
yup, I'll do that.

**Pull request merged**